minio/api.py

```diff
@@ -248,9 +248,11 @@ def make_bucket(self, bucket_name, location='us-east-1'):
         """
         is_valid_bucket_name(bucket_name)
 
-        ## Region already set in constructor, validate if
-        ## caller requested bucket location is same.
+        # Default region for all requests.
+        region = 'us-east-1'
         if self._region:
+            region = self._region
+            # Validate if caller requested bucket location is same as current region
             if self._region != location:
```

Comment:
Is this check needed?

Reply:
> Is this check needed?

This comes from the old code but also it seems to me it is correct to do the check, because if the user explicitly sets a region when initializing Minio constructor, most likely all buckets that he needs to create will be in the same region.